content/post/kallax.md

```diff
@@ -0,0 +1,218 @@
+
+Of course, we can make our own conditions by hand.
+
+```go
+q := NewUserQuery().
+        Where(kallax.Like(Schema.User.Username, "john%")).
+        Where(kallax.Lt(Schema.User.CreatedAt, time.Now().Add(-1 * time.Hour)))
+```
+
+Instead of using `"username"` to specify the column name we use the `Schema`, which is also generated.
+
+The way to access a field in the schema is just like you'd access a field in a struct.
+
+```
+Schema.$ModelName.$FieldName.$AnotherField.$YetAnotherField...
+```
+
+If a field is a struct, you can access its properties from the field as well. That's used for querying JSON.
+
+The right way to use the `Schema` and the operators is that you should be adding your custom `FindBy`s and then call them when using your queries. That way, all the conditions are in one place.
+
+We can add our method directly in the same file where our model was defined.
+
+```go
+func (q *UserQuery) FindByUsernameLike(pattern string) *UserQuery {
+        return q.Where(kallax.Like(Schema.User.Username, pattern))
+}
+```
+
+**WARNING:** define these in your own model file, never on the `kallax.go`. You should never edit that file, as it will be nuked every time you regenerate your models.
+
+## Dealing with relationships
+
+Consider the following models. We have a people, which can have many pets and a pet has an inverse relationships with its owner.
+```go
+type Person struct {
+        kallax.Model `table:"people"`
+        ID   int64   `pk:"autoincr"`
+        Name string
+        Pets []*Pet
+}
+
+type Pet struct {
+        kallax.Model  `table:"pets"`
+        ID    int64   `pk:"autoincr"`
+        Name  string
+        Kind  PetKind
+        Owner *Person `fk:"owner_id,inverse"`
+}
+
+type PetKind byte
+
+const (
+        Cat PetKind = iota
+        Dog
+        Fish
+)
+```
+
+On a sidenote, see the `PetKind` type? You'd think if you were using any other ORM or plain SQL you'd be able to use it. Truth is, you can't because it does not implement neither `sql.Scanner` nor `driver.Valuer`. Seems like a very silly thing, but it's a very common Go idiom to have such enum types. Because kallax generates code, it can wrap it and treat it as `byte` directly. So it's possible to use such types in kallax.
+
+So, let's find some people and their pets.
+
+```go
+store := NewPersonStore(db)
+
+q := NewPersonQuery().
+        FindByName("Steven").
+        WithPets(nil) // you can also pass a condition if you want pets filtered
+person, err := store.FindOne(q)
+```
+
+Kallax generated methods in your query to automatically preload your relationships. Just use `With{FieldName}`.
+Note that relationships are **not** preloaded by default, they must be explicitly preloaded using such methods.
+**WARNING:** preloading retrieves **all** the models of the relationship matching the giving condition, or just all of them if none was given. If the N side of your 1:N relationship is really big you may want to be doing the query from the other side.
```

Comment:
s/you may want to be doing the query from the other side/you may want to query from the other side/

Still not like it, but it is a little clearer.